File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        details = {}

        for match in matches:
            details.setdefault(match[0], 0)
            details[match[1]] = details.get(match[1], 0) - 1

        vec1 = []
        vec2 = []

        for key, value in details.items():
            if value == 0:
                vec1.append(key)
            elif value == -1:
                vec2.append(key)
        
        return [sorted(vec1), sorted(vec2)]
            

File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int, int> playersWinningDetails;

        for(auto it: matches) {
            playersWinningDetails[it[0]];
            playersWinningDetails[it[1]]--;
        }

        vector<vector<int>> ans(2);
        for(auto it: playersWinningDetails) {
            if(it.second == 0)
                ans[0].push_back(it.first);
            else if(it.second == -1)
                ans[1].push_back(it.first);
        }


        return ans;
    }
};